subsys/bluetooth/audio/vocs.c

```diff
@@ -64,6 +64,11 @@ static ssize_t write_location(struct bt_conn *conn, const struct bt_gatt_attr *a
 	}
 
 	memcpy(&inst->srv.location, buf, len);
+
+	if (inst->srv.location <= 0 || inst->srv.location > 0x08000000) {
+		inst->srv.location = old_location;
+	}
+
```

Comment:
I would suggest to avoid memcpy and simply check the buf content before it

Reply:
Done